### PR TITLE
Fix definitons

### PR DESCRIPTION
**Pull request by kaorun343**
* add “FunctionalComponentOptions” and “ContextObject”
* fix and update types

#3509 and #3582

### CODE REVIEW

File: types/options.d.ts [hunk 1]
```diff
@@ -39,6 +41,21 @@ export interface ComponentOptions {
...
 }
 
+export interface FunctionalComponentOptions {
+  props?: string[] | { [key: string]: PropOptions | Constructor | Constructor[] };
+  functional: boolean;
+  render(this: never, createElement: $createElement, context: ContextObject): VNode;
+  name?: string;
+}
+
+export interface ContextObject {
```
Comment (ktsn) [inline types/options.d.ts]: I think it would be better to rename this interface to more concrete name, like `RenderContext`?

**Comment by ktsn** [comment]
@kaorun343 Thanks for your quick fix 😄

**Comment by yyx990803** [comment]
Seems that there are conflicts after merging #3582 - can we resolve that?

**Comment by kaorun343** [comment]
I will resolve that.

**Pull request merged**